samza-kafka/src/main/scala/org/apache/samza/system/kafka_deprecated/KafkaSystemConsumer.scala

```diff
@@ -283,15 +284,16 @@ private[kafka_deprecated] class KafkaSystemConsumer(
         null
       }
 
-      if(fetchLimitByBytesEnabled ) {
-        val ime = new IncomingMessageEnvelope(systemStreamPartition, offset, key, message, getMessageSize(msg.message))
-        ime.setTimestamp(if (!msg.message.isNull) msg.message.timestamp else 0L)
-        put(systemStreamPartition, ime)
-      } else {
-        val ime = new IncomingMessageEnvelope(systemStreamPartition, offset, key, message)
-        ime.setTimestamp(if (!msg.message.isNull) msg.message.timestamp else 0L)
-        put(systemStreamPartition, ime)
-      }
+      val eventTime = if (!msg.message.isNull) msg.message.timestamp else 0L
```

Comment:
This looks like a deprecated code. I am not sure whether we really need to update this.

Reply:
it is deprecated code, the setTimestamp API does no longer exist